auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java

```diff
@@ -377,14 +378,19 @@ public MethodSpec createWriteMethod(TypeName autoValueClassName,
         .addException(IOException.class);
 
     writeMethod.addStatement("$N.beginObject()", writer);
+
+    NameAllocator nameAllocator = new NameAllocator();
```

Comment:
You need to add "writer" and "value" to this before the code below runs. Please add a test that uses these two names as properties.